src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -115,6 +116,11 @@ public JsonSerializerOptions(JsonSerializerOptions options)
             _propertyNameCaseInsensitive = options._propertyNameCaseInsensitive;
             _writeIndented = options._writeIndented;
 
+            if (_serializerContext == null && _typeInfoResolver != s_defaultTypeInfoResolver)
+            {
+                _typeInfoResolver = options._typeInfoResolver;
+            }
```

Comment:
There is a bit of gymnastics here which I don't like but that's what we need to do in order to avoid breaking changes. This is one of the things we should consider changing (but separately so that it's easier to see breaking changes)

Reply:
I suspect this might be simplified if the two fields get merged, as discussed elsewhere.